ddpui/core/trial/clone_service.py

```
@@ -0,0 +1,225 @@
+        trialclone.save(update_fields=["status", "error", "updated_at"])
+        logger.error(f"clone {trialclone.id} failed: {err}")
+        # best-effort teardown of whatever got created before the failure — never let a
+        # teardown problem mask the original exception, which must still propagate.
+        try:
+            if trialclone.trial_org:
+                logger.info(f"tearing down org+workspace for failed clone {trialclone.id}")
+                OrgCleanupService(trialclone.trial_org, dry_run=False).delete_org()
+            if trialclone.manifest.get("trial_warehouse_db"):
+                logger.info(f"dropping trial database for failed clone {trialclone.id}")
+                drop_trial_database(trialclone.trial_email)
+        except Exception as cleanup_err:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)